src/mgmt/srv_pool.c

```diff
@@ -58,6 +58,23 @@ ds_mgmt_tgt_pool_destroy_ranks(uuid_t pool_uuid, d_rank_list_t *filter_ranks)
 	return rc;
 }
 
+static uint32_t
+pool_create_rpc_timeout(crt_rpc_t *tc_req, size_t scm_size)
+{
+	uint32_t	timeout;
+	uint32_t	extra_timeout = 0;
+	int		rc;
+
+	rc = crt_req_get_timeout(tc_req, &timeout);
+	D_ASSERTF(rc == 0, "crt_req_get_timeout: "DF_RC"\n", DP_RC(rc));
+	if (bio_nvme_configured(SMD_DEV_TYPE_META)) {
+		extra_timeout = scm_size / ((size_t)1024 * 1024 * 1024 * 5);
+		extra_timeout = min(60, extra_timeout);
+	}
+
+	return timeout + extra_timeout;
```

Comment:
I think a proper way is to estimate timeout based on pool size, then return the max(default, estimated).